api/login.ts

```diff
@@ -88,16 +89,41 @@ export default async function handler(req: VercelRequest, res: VercelResponse) {
     if (!profile) {
       const { insert_profiles_one } = await adminClient.mutate(
         {
-          insert_profiles_one: [{ object: { address: address } }, { id: true }],
+          insert_profiles_one: [
+            { object: { address: address } },
+            {
+              id: true,
+              users: [
+                {},
+                {
+                  circle_id: true,
+                },
+              ],
+            },
+          ],
         },
         {
           operationName: 'login_insertProfile',
         }
       );
+      assert(insert_profiles_one, "panic: adding profile didn't succeed");
       profile = insert_profiles_one;
+      await insertInteractionEvent({
+        event_type: 'first_login',
+        profile_id: insert_profiles_one.id,
+        circle_id: insert_profiles_one.users?.[0]?.circle_id,
+        data: {
+          invitedMember: insert_profiles_one.users
```

Comment:
since these two props are exclusive, maybe what we really want is a boolean e.g. `brandNew`? 

actually i'm a bit confused by the logic here; is there a non-exceptional case when these should both be false? should it instead be checking `(insert_profiles_one.users?.length || 0)` instead of using a ternary, so one of these is always true?

Reply:
ok i collapsed this to just `brandNew`